tests/Replication2/Streams/TestLogSpecification.h

```diff
@@ -0,0 +1,109 @@
+#include <Replication2/Mocks/PersistedLog.h>
+#include <Replication2/Mocks/ReplicatedLogMetricsMock.h>
+
+namespace arangodb::replication2::test {
+
+struct LogMultiplexerTestBase : ::testing::Test {
+  static auto createReplicatedLog(LogId id = LogId{0})
+      -> std::shared_ptr<replication2::replicated_log::ReplicatedLog> {
+    return createReplicatedLogImpl<replicated_log::ReplicatedLog, test::MockLog>(id);
+  }
+
+  static auto createAsyncReplicatedLog(LogId id = LogId{0})
+      -> std::shared_ptr<replication2::replicated_log::ReplicatedLog> {
+    return createReplicatedLogImpl<replicated_log::ReplicatedLog, test::AsyncMockLog>(id);
+  }
+
+  static auto createFakeReplicatedLog(LogId id = LogId{0})
+      -> std::shared_ptr<replication2::test::TestReplicatedLog> {
+    return createReplicatedLogImpl<replication2::test::TestReplicatedLog, test::MockLog>(id);
+  }
+
+ private:
+  template <typename Impl, typename MockLog>
+  static auto createReplicatedLogImpl(LogId id) -> std::shared_ptr<Impl> {
+    auto persisted = std::make_shared<MockLog>(id);
+    auto core = std::make_unique<replicated_log::LogCore>(persisted);
+    auto metrics = std::make_shared<ReplicatedLogMetricsMock>();
+    return std::make_shared<Impl>(std::move(core), metrics,
+                                  LoggerContext(Logger::REPLICATION2));
+  }
+};
+
+struct default_deserializer {
+  template <typename T>
+  auto operator()(streams::serializer_tag_t<T>, velocypack::Slice s) -> T {
+    return s.extract<T>();
+  }
+};
+
+struct default_serializer {
+  template <typename T>
+  void operator()(streams::serializer_tag_t<T>, T const& t, velocypack::Builder& b) {
+    b.add(velocypack::Value(t));
+  }
+};
+
+inline constexpr auto my_int_stream_id = streams::StreamId{1};
+inline constexpr auto my_string_stream_id = streams::StreamId{8};
+inline constexpr auto my_string2_stream_id = streams::StreamId{9};
+
+inline constexpr auto my_int_stream_tag = streams::StreamTag{12};
+inline constexpr auto my_string_stream_tag = streams::StreamTag{55};
+inline constexpr auto my_string2_stream_tag = streams::StreamTag{56};
+inline constexpr auto my_string2_stream_tag2 = streams::StreamTag{58};
+
+/* clang-format off */
+
+using MyTestSpecification = streams::stream_descriptor_set<
+      streams::stream_descriptor<my_int_stream_id, int, streams::tag_descriptor_set<
+        streams::tag_descriptor<my_int_stream_tag, default_deserializer, default_serializer>
+      >>,
+      streams::stream_descriptor<my_string_stream_id, std::string, streams::tag_descriptor_set<
+        streams::tag_descriptor<my_string_stream_tag, default_deserializer, default_serializer>
+      >>,
+      streams::stream_descriptor<my_string2_stream_id, std::string, streams::tag_descriptor_set<
+        streams::tag_descriptor<my_string2_stream_tag, default_deserializer, default_serializer>,
+        streams::tag_descriptor<my_string2_stream_tag2, default_deserializer, default_serializer>
```

Comment:
The second tag doesn't seem to be in use, or is it?

Reply:
Well, it is somewhat part of the test. Imagine a multiplexer that would crash when multiple tags are involved or could only detect the last tag in the list.